TFT/src/User/API/Gcode/mygcodefs.c

```diff
@@ -71,16 +71,22 @@ bool scanPrintFilesGcodeFs(void)
       line++;            // (use a common file path) e.g. "/sub_dir/cap.gcode" -> "sub_dir/cap.gcode"
 
     // "line" never has "/" at the beginning of a path (e.g. "sub_dir/cap.gcode") while "infoFile.title" has it
-    // (e.g. "bSD:/sub_dir"), so we skip it during the check of current directory match (index 5 used instead of 4)
-    if (strlen(infoFile.title) > 5 && strstr(line, infoFile.title + 5) == NULL)  // if "line" doesn't include current directory
-      continue;
+    // (e.g. "bSD:/sub_dir"), so we skip it during the check of current folder match (index 5 used instead of 4)
+    Pstr_tmp = strstr(line, infoFile.title + 5);  // "+ 5" skips the 5 bytes related to prefix "bSD:/" in infoFile.title
```

Comment:
you are accessing infoFile.title + 5 before checking the length for infoFile.title!

Reply:
Thank you. Fixed here: https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/pull/2291/commits/0d5297c5411ec741f5b842bc800584e8cdbe78bf